engineering_tools/monitor-ros-cpu.py

```diff
@@ -15,17 +46,61 @@ def parse_args():
         default="carma-cpu-usage-logs",
         help="Directory to store output files (default: carma-cpu-usage-logs)",
     )
+    parser.add_argument(
+        "--include-pattern",
+        "-i",
+        help="Additional comma-separated patterns to include in process filtering",
+    )
     return parser.parse_args()
 
+
 def setup_logging_directory(output_dir):
     if not os.path.exists(output_dir):
         os.makedirs(output_dir)
     timestamp = datetime.now().strftime("%Y_%m_%d-%H_%M_%S")
     filename = f"cpu_usage_ros2_nodes_{timestamp}.csv"
     return os.path.join(output_dir, filename)
 
+
+def is_ros_related_process(proc_info, cmdline):
+    """
+    Check if a process is ROS-related based on name and command line
+    """
+    # Convert process information to lowercase for case-insensitive matching
+    name_lower = proc_info["name"].lower()
+    cmdline_lower = cmdline.lower()
+
+    # Check exclusions first
+    if any(excl in name_lower or excl in cmdline_lower for excl in EXCLUDE_KEYWORDS):
```

Comment:
If there is an excluded keyword in the process and also a ROS keyword this would still return false, is that the desired behavior?

Reply:
Yeah that is the behavior. Current keywords are picked just like that, but I included a note on the comment to also emphasize that.